### ISSUE TITLE
Backspace does not work with Safari

### ISSUE THREAD

**Issue by mstahv** [miztroh/wysiwyg-e#215]
If you try to hit backspace with Safari in the the text area, the text is not erased.

**Comment by pcornelissen** [comment]
Halloweenman in the polymer slack channel had the same problem

**Comment by miztroh** [comment]
Will investigate.  Thanks for the report!

**Comment by miztroh** [comment]
Spoke too soon.  I fixed one issue, but there appear to be more.

**Comment by Icer2000** [comment]
Any news on this? The backspace also does not work properly in Firefox. After using it as first command, all future keydowns are wrapped in <p> tags, so that no writing of text is possible. Sample screenshot:
![image](https://user-images.githubusercontent.com/16884132/39925555-6b719f22-552c-11e8-9ef6-45cc81cd5955.png)

**Comment by miztroh** [comment]
@Icer2000: Yeah, I've got a line on a Macbook Air I can use for development so that I can get some proper Safari testing.  ETA on that is probably ~3 weeks, so please bear with me on that.

Regarding the Firefox issue, is this on Mac only or can you reproduce on Windows as well.

**Comment by Icer2000** [comment]
I tried to use it on Firefox for Windows.

I also tested in IE11 and EDGE on Windows and had issues with backspace
too. It seems the changes that you reverse Integrated from the 3.x branch
seem to have broken some stuff. :)

**Comment by miztroh** [comment]
I was able to confirm the Firefox issue.  Interesting that it doesn't seem to happen on Chrome.  I will investigate.  Thanks for the report.

**Comment by miztroh** [comment]
I've tested Firefox, Edge, and IE11.  They all work as of the latest release.

@mstahv:  Can I get you to test the latest release and let me know if this is still an issue?  If so, I should have a Macbook Air available sometime in the next couple weeks to test and resolve this issue.  Thank you!

**Comment by miztroh** [comment]
I have also backported the fixes from the 3.x (master) branch to the 2.x branch and created a new 2.x release.

https://github.com/miztroh/wysiwyg-e/releases/tag/v2.1.4

**Comment by pleku** [comment]
I've just tested with component version 2.1.4 and Safari 11.1 on macOS High Sierra, and normal backspacing still doesn't work. I can only remove text after the cursor by pressing "delete" (fn + backspace) or by painting text and then pressing delete. Just backspace still does nothing.

**Comment by miztroh** [comment]
I've updated the issue title to reflect that the issue is now localized to just Safari.  I've tried to troubleshoot via Sauce Labs, but that's been a miserable experience.  I've got a Macbook Air on the way.  ETA ~ 2 weeks.  Then I'll dig into this issue.  Thanks for your patience.

**Comment by Icer2000** [comment]
Any news? :)

**Comment by bboyrabi** [comment]
This issue has not been solved yet.
I tried version 2.1.4 on Safari 11.1 on macOS High Sierra, but backspacing and toolbox does not work.
Sahdow root in safari seems to have no getSelection function.
Is there a solution to this?
I thought I could solve it by using [shadow-selection-polyfill](https://github.com/GoogleChromeLabs/shadow-selection-polyfill), but I have not tried it yet.
Or is there any other way?
Thank you.
